### PR TITLE
Update https-utilities

### PR DESCRIPTION
**Pull request by dqos**
Add CommonName to self-signed certificate to fix: https://github.com/nextcloud-snap/nextcloud-snap/issues/1075

### CODE REVIEW

File: src/https/utilities/https-utilities [hunk 1]
```diff
@@ -36,7 +36,7 @@ generate_self_signed_certificate()
 	chmod 750 "$(dirname "$SELF_SIGNED_CHAIN")"
 
 	openssl req -newkey rsa:4096 -nodes -keyout "$SELF_SIGNED_KEY" \
-	        -x509 -days 90 -out "$SELF_SIGNED_CERT" -subj "/O=Nextcloud"
+	        -x509 -days 90 -out "$SELF_SIGNED_CERT" -subj "/O=Nextcloud/CN=Nextcloud"
```
Comment (kyrofa) [inline src/https/utilities/https-utilities:39]: You literally only need a CN entry? It doesn't even matter what is is?! This seems broken: the CN is supposed to be the domain name you're using to visit the site protected by the cert.

Comment (dqos) [reply src/https/utilities/https-utilities:39]: Yes, this fixes it in order to mount webdav drives. For example davfs2 complaints about a missing CN:
```
/sbin/mount.davfs: Mounting failed.
Server certificate was missing commonName attribute in subject name```

The CN value is actually deprecated, browsers use the SAN value. But for some reason, davfs2 checks the CN. I think in order to do it properly, we can fill in the current hostname of the server.

Comment (dqos) [reply src/https/utilities/https-utilities]: I have adjusted the CN to be dynamic now based on the hostname. Feel free to merge or close, it fixed it for our completely internal setup 😄

**Review by scubamuc — APPROVED** [review]

**Review by kyrofa — CHANGES_REQUESTED** [review]
I don't think we want this as it stands today. If we want to make the CN configurable, we can do that, but it'll be a larger feature through the config hook. If you want to take a crack at it I can help guide you.

**Pull request closed without merging**